src/common/token/baseEth.ts

```diff
@@ -33,4 +33,46 @@ export class BaseEthToken extends EthereumToken {
       pollingOptions,
     });
   }
+
+  public async pollForTxBeingAvailable({
+    txId,
+  }: {
+    txId: string;
+  }): Promise<void> {
+    this.logger.debug('Polling for transaction to be available on chain', {
+      txId,
+      ...this.pollingOptions,
+    });
+    await new Promise((resolve) =>
+      setTimeout(resolve, this.pollingOptions.initialBackoffMs),
+    );
+
+    let attempts = 0;
+    while (attempts < this.pollingOptions.maxAttempts) {
+      try {
+        const tx = await this.rpcProvider.getTransactionReceipt(txId);
+
+        if (tx) {
+          const confirmations = await tx.confirmations();
+          if (confirmations >= 1) {
+            this.logger.debug('Transaction found on chain', {
+              txId,
+              tx,
+              confirmations,
+            });
+            return;
+          }
```

Comment:
Consider debug logging the case where tx was undefined. Also add attempt count to your debug logs.